streaming/rrdpush.c

```diff
@@ -676,178 +661,307 @@ int rrdpush_receiver_thread_spawn(struct web_client *w, char *url) {
         if(!name || !*name) continue;
         if(!value || !*value) continue;
 
-        if(!strcmp(name, "key"))
-            key = value;
-        else if(!strcmp(name, "hostname"))
-            hostname = value;
-        else if(!strcmp(name, "registry_hostname"))
-            registry_hostname = value;
-        else if(!strcmp(name, "machine_guid"))
-            machine_guid = value;
+        if(!strcmp(name, "key") && !rpt->key)
+            rpt->key = strdupz(value);
+
+        else if(!strcmp(name, "hostname") && !rpt->hostname)
+            rpt->hostname = strdupz(value);
+
+        else if(!strcmp(name, "registry_hostname") && !rpt->registry_hostname)
+            rpt->registry_hostname = strdupz(value);
+
+        else if(!strcmp(name, "machine_guid") && !rpt->machine_guid)
+            rpt->machine_guid = strdupz(value);
+
         else if(!strcmp(name, "update_every"))
-            update_every = (int)strtoul(value, NULL, 0);
-        else if(!strcmp(name, "os"))
-            os = value;
-        else if(!strcmp(name, "timezone"))
-            timezone = value;
-        else if(!strcmp(name, "abbrev_timezone"))
-            abbrev_timezone = value;
+            rpt->config.update_every = (int)strtoul(value, NULL, 0);
```

Comment:
Is there a specific reason to unse `strtoul` instead our [str2ul](https://github.com/netdata/netdata/blob/3b2e7553be07d1eada4842694ba6ec95e696018c/libnetdata/inlined.h#L113-L123)?